finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.{Buffer, BufferWriter}
+
+trait CanBeParameter[-A] { outer =>
```

Comment:
I'm not sure we should have the variance either. Even if we have a X <: Y relationship, we should still be forced to think about how that serializes into a Parameter (e.g. sizeOf, typeCode, etc. are likely to be different).